packages/controllers/src/utils.ts

```diff
@@ -1,5 +1,40 @@
 import { Timer } from './snaps/Timer';
 
+declare interface AssertionError extends Error {
+  code: 'ERR_ASSERTION';
+}
+let _assert: (value: any, message?: string | Error) => asserts value;
+let _AssertionError: new (options: { message: string }) => AssertionError;
+
+/* istanbul ignore next */
+if (typeof window === 'undefined') {
+  // eslint-disable-next-line @typescript-eslint/no-require-imports
+  ({ _assert, _AssertionError } = require('assert'));
```

Comment:
We never hit this case in our tests because we test with `jsdom`, but this will never assign a value to the variables anyway. This can be tested by running the following JS snippet locally:
```js
let _assert, _AssertionError;
({ _assert, _AssertionError } = require('assert'));

console.log(_assert, _AssertionError); // undefined, undefined
```

I'm not sure if we want to have a case where we fall back to the Node.js assert. It may just be better to roll our own `assert` in all cases, as you do in the `else` case. Moreover, if we want to pursue this pattern, the implementation of `assert` should be migrated to `@metamask/utils`.

Reply:
Waiting for re-deploy of `@metamask/utils` takes too long, xstate has been in dev for a few now and I'd like to get it merged as soon as possible.

I've moved it to snap-utils